src/zeek.bif

```diff
@@ -4080,6 +4080,25 @@ function lookup_hostname%(host: string%) : addr_set
 	return nullptr;
 	%}
 
+## Issues a synchronous DNS lookup.
+##
+## host: The hostname to lookup.
+##
+## Returns: A set addresses, either IPv4 or IPv6, associated with *host*.
+##
+## .. zeek:see:: lookup_addr
+##
+## .. note::
+##
+##      This is a blocking call. You should use :zeek:see:`lookup_hostname`
+##      unless for initialization or testing purposes.
+##
+## .. zeek:see:: lookup_addr lookup_hostname
+function lookup_hostname_sync%(host: string%) : addr_set
```

Comment:
Nit: like discussed offline we could use another name for this: #4163 seems to go with dedicated namespaces for async/blocking versions, but for this one we could just go to a truly ugly name since using this is a red flag, e.g., `lookup_hostname_blocking` or similar.

Reply:
I'll go with `blocking_lookup_hostname`, as we have `lookup_hostname_txt` with another suffix. Looks scary enough :-)